src/android/io/sentry/SentryCordova.java

```diff
@@ -1,47 +1,370 @@
+        }
+        if (jsonOptions.has("maxBreadcrumbs")) {
+          options.setMaxBreadcrumbs(jsonOptions.getInt("maxBreadcrumbs"));
+        }
+        if (jsonOptions.has("environment") && jsonOptions.getString("environment") != null) {
+          options.setEnvironment(jsonOptions.getString("environment"));
+        }
+        if (jsonOptions.has("release") && jsonOptions.getString("release") != null) {
+          options.setRelease(jsonOptions.getString("release"));
+        }
+        if (jsonOptions.has("dist") && jsonOptions.getString("dist") != null) {
+          options.setDist(jsonOptions.getString("dist"));
+        }
+        if (jsonOptions.has("enableAutoSessionTracking")) {
+          options.setEnableSessionTracking(jsonOptions.getBoolean("enableAutoSessionTracking"));
+        }
+        if (jsonOptions.has("sessionTrackingIntervalMillis")) {
+          options.setSessionTrackingIntervalMillis(jsonOptions.getInt("sessionTrackingIntervalMillis"));
+        }
+        if (jsonOptions.has("enableNdkScopeSync")) {
+          options.setEnableScopeSync(jsonOptions.getBoolean("enableNdkScopeSync"));
+        }
+        if (jsonOptions.has("attachStacktrace")) {
+          options.setAttachStacktrace(jsonOptions.getBoolean("attachStacktrace"));
+        }
+        if (jsonOptions.has("attachThreads")) {
+          // JS use top level stacktraces and android attaches Threads which
+          // hides them so by default we hide.
+          options.setAttachThreads(jsonOptions.getBoolean("attachThreads"));
+        }
 
-          SentryAndroid.init(this.cordova.getActivity().getApplicationContext(), options -> {
-            options.setDsn(dsn);
-          });
+        if (jsonOptions.has("enableNativeCrashHandling") && !jsonOptions.getBoolean("enableNativeCrashHandling")) {
+          final List<Integration> integrations = options.getIntegrations();
+          for (final Integration integration : integrations) {
+            if (integration instanceof UncaughtExceptionHandlerIntegration || integration instanceof AnrIntegration || integration instanceof NdkIntegration) {
+              integrations.remove(integration);
+            }
+          }
+        }
 
-          callbackContext.sendPluginResult(new PluginResult(Status.OK, true));
-          break;
-        default:
-          callbackContext.sendPluginResult(new PluginResult(Status.ERROR, "not implemented"));
-          break;
+        logger.info(String.format("Native Integrations '%s'", options.getIntegrations().toString()));
+        sentryOptions = options;
+      } catch (JSONException e) {
+        logger.info("Error parsing options JSON sent over native bridge.");
+        callbackContext.sendPluginResult(new PluginResult(Status.ERROR, false));
       }
+    });
 
-      return true;
+    callbackContext.sendPluginResult(new PluginResult(Status.OK, true));
+  }
+
+  private void captureEnvelope(String envelope, final CallbackContext callbackContext) {
```

Comment:
I don't see much benefit in `captureEnvelope` with a `String` envelope.
We need to make a bridge call to be able to get `bytes.length` from the payload anyway to build the envelope on the JS side.

Why don't we simply pass the event and create the envelope in here?

So a `captureJsonEvent`?

Reply:
it'd need to be an Array<String> then anyway because the header is built by the JS layer otherwise the Android bits would need to deserialize/serialize again.
Also, an envelope could be more than just an event, so captureJsonEvent would not work either (naming).

Reply:
> it'd need to be an Array then anyway because the header is built by the JS layer otherwise the Android bits would need to deserialize/serialize again (you can't pass a complex object afaik).

We can still pass a string, but instead of an envelope string, it's just the payload as a string. Inside we can calculate the size (since the code lives in the Java layer anyway, so we avoid that bridge call) and wrap the thing into an envelope.

> Also, an envelope could be more than just an event, so captureJsonEvent would not work either (naming).

But it's not. Currently we capture only a single event.

Reply:
yeah makes sense, on Flutter we don't use the Native layer to calculate the event's `length`, we pass the String already prepared to be cached in the disk, because Dart has an API for utf8 encoding

Reply:
Implemented in #208